### PR TITLE
feat(EditorSource): add support for POST requests

### LINKED ISSUE #1788: Add HTTP-POST support

### PR DESCRIPTION
**Pull request by gabaldon**
This pr add support in Sheikah for POST kind sources. This must be tested with [witnet-radon-js #85](https://github.com/witnet/witnet-radon-js/pull/85).
close #1788

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -188,7 +188,17 @@ export function encodeDataRequest(radRequest) {
     timeLock: radRequest.timelock,
     time_lock: radRequest.timelock,
     retrieve: radRequest.retrieve.map(retrieve => {
-      return { ...retrieve, script: [...cbor.encode(retrieve.script)] }
+      // Add support for utf16
```
Comment (Tommytrg) [inline src/utils.js:191]:
```suggestion
      // TODO: Add support for utf16
```

File: src/components/EditorSource.vue [hunk 2]
```diff
@@ -122,68 +166,33 @@ export default {
...
           index: this.index,
           source: {
-            protocol: this.localProtocol.primaryText,
-            url: this.localUrl,
-            contentType: val.primaryText,
+            protocol: this.form.localProtocol.primaryText,
+            headers: isValidJson(this.form.localHeaders)
+              ? JSON.parse(this.form.localHeaders)
+              : {},
+            body: isValidJson(this.form.localBody) ? this.form.localBody : {},
```
Comment (Tommytrg) [inline src/components/EditorSource.vue:180]: The body doesn't need to be a json

File: src/utils.js [hunk 3]
```diff
@@ -188,7 +188,17 @@ export function encodeDataRequest(radRequest) {
...
+      let utf8Encode = new TextEncoder()
+      return {
+        ...retrieve,
+        headers: Object.entries(retrieve.headers),
+        // Only add body if it is a POST request
```
Comment (Tommytrg) [inline src/utils.js:196]:
```suggestion
        // Only add body if it exists
```

File: src/components/EditorSource.vue (hunk 2 above)
Comment (gabaldon) [reply src/components/EditorSource.vue:180]: but it needs to be valid as a json

File: src/components/EditorSource.vue [hunk 4]
```diff
@@ -99,10 +128,24 @@ export default {
     },
   },
   data() {
+    const isValidObject = (rule, value, callback) => {
```
Comment (Tommytrg) [inline src/components/EditorSource.vue:131]:
```suggestion
    const areValidHeaders = (rule, value, callback) => {
```

**Review by Tommytrg — APPROVED** [review]

**Pull request merged**